app/controllers/messages_controller.rb

```diff
@@ -1,6 +1,17 @@
 class MessagesController < ApplicationController
   def create
+    @user = User.find_or_create_by(fb_id: fb_id_param)
+    if @user
+      @message = @user.messages.new(json_data: params[:entry])
```

Comment:
consider switching to just saving the necessary fields, i.e. `text`, `sender_fb_id`, `recipient_fb_id`